src/viam/app/viam_client.py

```diff
@@ -179,3 +182,47 @@ def close(self):
         LOGGER.debug("Closing gRPC channel to app.")
         self._channel.close()
         self._closed = True
+
+    async def connect_to_machine(self, *, address: Optional[str] = None, id: Optional[str] = None) -> RobotClient:
+        """Connect to a machine using existing credentials.
+
+        A connection can be attempted using either the machine's address or its ID.
+        If both an address and ID are provided, the address will take precedence and the ID will be ignored.
+
+        ::
+
+            async def connect() -> ViamClient:
+                # Replace "<API-KEY>" (including brackets) with your API key and "<API-KEY-ID>" with your API key ID
+                dial_options = DialOptions.with_api_key("<API-KEY>", "<API-KEY-ID>")
+                return await ViamClient.create_from_dial_options(dial_options)
+
+
+            async def main():
+                viam_client = await connect()
+
+                # Connect to a machine and obtain a RobotClient
+                # Replace "<MACHINE_ADDRESS>" (including brackets) with your machine's connection address
+                machine = await viam_client.connect_to_machine(address="<MACHINE_ADDRESS>")
+
+        Args:
+            address (Optional[str]): The address (FQDN) of the machine. Defaults to None.
+            id (Optional[str]): The ID (as a UUID) of the machine. Defaults to None.
+
+        Raises:
+            ValueError: If neither an address nor ID is provided.
+
+        Returns:
+            RobotClient: The active connection to the machine.
+        """
+        if address is None and id is None:
+            raise ValueError("Either a machine address or ID must be provided")
+
+        if id is not None and address is None:
+            parts = await self.app_client.get_robot_parts(id)
+            main_part = [p for p in parts if p.main_part][0]
```

Comment:
I don't know how main parts work; are there multiple main parts with the same address?

Reply:
There should be one main part per machine, and this will find it.